### ISSUE TITLE
ui.Delay - named or something (suggestion)

### ISSUE THREAD

**Issue by Phuket2** [omz/Pythonista-Issues#79]
ui.Delay is so nice if you don't want to play around with threads. But the fact that as far as I can tell ui.Delay, ui.cancel_delays are global, it's usage is very limited. Example, would be not smart to use it in any type of share code. If the caller of your share code also uses ui.Delay, it will go wrong.  
But if you could say for example ui.Delay(foo.bar, 1, named = 'my_delay') and then had to do that ui.cancel_delays('my_delay') meaning explicitly clear it.
I am Not sure if this is difficult or not. But It seems to me that a implementation simlar to this would achieve something closer to being supportive. 
I don't think I have articulated this really clearly, but I think my meaning is still clear.

**Comment by omz** [comment]
Yes, this is definitely a "problem area". Another (related) thing that might be useful would be some way to attach a repeating timer to a particular view, i.e. in a way that the timer only fires while the view is visible. That would make it easier to implement things like clocks, countdown labels, and the like.

**Comment by Phuket2** [comment]
That's a great idea. But rather than attach a timer you could just add a timer style object to ui.View. Hmmm, well maybe not. I am not sure if you currently add any attrs to a ui.View that are not in the objective-C version of UI.View. But regardless, it's a good thought

**Comment by Phuket2** [comment]
@omz, if you did something like having a timing attached to the ui to fire on an interval, would be good if you could allow it to fire even if the view was hidden.  Would allow for easy way to do some background tasks. Maybe web lookups etc...I guess it depends if your implementation would be blocking or not I guess. But regardless just the firing would be nice. Not sure if you would do it like an action style or a callback into the ui.View delegate so to speak.

**Comment by controversial** [comment]
For anyone who wants this, this is implemented in `ui2`. See https://github.com/controversial/ui2/blob/master/ui2/delay.py

I might add a timer like @omz suggested soon.
